ansible_collections/f5networks/f5_modules/plugins/modules/bigip_device_group.py

```diff
@@ -242,6 +269,14 @@ def auto_sync(self):
             return True
         return False
 
+    @property
+    def asm_sync(self):
+        if self._values['asm_sync'] is None:
+            return None
+        elif self._values['asm_sync'] == 'enabled':
+            return True
+        return False
+
```

Comment:
remove this its not needed

Reply:
The `asm_sync` properties is quite similar to `auto_sync` in its behaviour within the API, hence this is required. Removing this leads to the module not being able to understand the API parameters and always starting to report changes for this setting.

Reply:
you are going by wrong examples, flatten boolean is for ensuring that if i specify for example: true, yes, True etc. in the playbook we make it either 'yes' and 'no' same goes for whatever api returns, if we see enabled/on/True etc we ensure internally we have it as 'yes' or 'no' this is done for comparison. Look at other modules how we handle it.

Reply:
Removing this leads to module failures as the API parameter would not be recognized anymore (same behaviour applies as for `auto_sync`/`autoSync`).
Refactoring to `flatten_boolean` seems like a good idea, other modules already use it, but this would be beyond the scope of this feature PR as it applies to all properties of class `ApiParameters`.

I think refactoring `flatten_boolean` should be considered for a separately.